app/utils/queryBuilder.ts

```diff
@@ -382,10 +397,18 @@ const buildQuery = ({
 };
 
 /**
- * Hook which memoizes the graphql query from the
- * provided queryConfig
+ * Hook which builds a memoized graphql query from the query configuration
+ * @param {QueryConfig} queryConfig - the QueryConfig object
+ * @param {Filter[]} contextFilters - an optional filter array to be included the
+ * query's `where` expression. It is expected that these filters would be set from
+ * an app context that is not wanted to be kepts in local storage, such as a
+ * URL query param
+ * @returns {string} a graphql querry
  */
-export const useQueryBuilder = (queryConfig: QueryConfig): string =>
+export const useQueryBuilder = (
+  queryConfig: QueryConfig,
+  contextFilters?: Filter[]
+): string =>
   useMemo(() => {
-    return buildQuery(queryConfig);
+    return buildQuery(queryConfig, contextFilters);
   }, [queryConfig]);
```

Comment:
i see the eslint error here about `contextFilters` being missing in the dependency array